File: IF-Coders/Aulas 04 e 05 - Caractere e String/charString02.py
"""
Escreva um programa que leia três strings.
A seguir imprimir as 3 strings em ordem alfabética.
COLETÂNEA DE EXERCÍCIOS E NOTAS DE AULA EM LINGUAGEM DE PROGRAMAÇÃO C
- Prof Dr David Buzatto (v1.0 - 2019).
"""
string1 = str(input("Digite a primeira string: "))
string2 = str(input("Digite a segunda string: "))
string3 = str(input("Digite a terceira string: "))
string1 = string1.lower()
string2 = string2.lower()
string3 = string3.lower()
print("\nEm ordem alfabética:\n")
if string1 < string2 < string3:
    print(string1, string2, string3)
elif string1 < string3 < string2:
    print(string1, string3, string2)
elif string2 < string1 < string3:
    print(string2, string1, string3)
elif string2 < string3 < string1:
    print(string2, string3, string1)
elif string3 < string1 < string2:
    print(string3, string1, string2)
else:
    print(string3, string2, string1)

File: IF-Coders/Aulas 04 e 05 - Caractere e String/charString02.c
/*
Escreva um programa que leia três strings.
A seguir imprimir as 3 strings em ordem alfabética.
COLETÂNEA DE EXERCÍCIOS E NOTAS DE AULA EM LINGUAGEM DE PROGRAMAÇÃO C - Prof Dr David Buzatto (v1.0 - 2019).
Autor: Zilli
*/
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

    int main() {

    char nome[3][41];
    char controle[41];
    int i;
    int j;
    int comparacao;//comparação = strcmp;

    for( i = 0; i < 3; i++ ) {

    printf("String %d: ",  i + 1 );
    fgets(nome[i], 41, stdin );
    nome[i] [ strlen( nome[i] )-1] = '\0';

    }

    for( i = 0; i < 3; i++ ) {

        for(j = i + 1; j < 3 ; j++) {//i + 1 (printf);

            comparacao = strcmp( nome[i], nome[j] );

            if( comparacao > 0 ) {

                strcpy( controle, nome[j] );//variável temporária;
                strcpy( nome[j], nome[i] );
                strcpy( nome[i], controle);

            }
        }
    }

    printf("\nEm ordem alfabetica: %s, %s e %s\n", nome[0], nome[1], nome[2] );

    return 0;

    }
